core/graph/src/main/java/uk/gov/gchq/gaffer/operation/export/graph/handler/ExportToOtherAuthorisedGraphHandler.java

```diff
@@ -0,0 +1,183 @@
+                .graphId(graphId)
+                .library(graphLibrary)
+                .build();
+    }
+
+    private Graph createGraphAfterResolvingParentSchemaAndProperties(final User user,
+                                                                     final GraphLibrary graphLibrary,
+                                                                     final String graphId,
+                                                                     final String parentSchemaId,
+                                                                     final String parentStorePropertiesId) {
+        final Schema schema = resolveSchema(user, graphLibrary, parentSchemaId);
+        final StoreProperties storeProperties = resolveStoreProperties(user, graphLibrary, parentStorePropertiesId);
+
+        return new Graph.Builder()
+                .graphId(graphId)
+                .library(graphLibrary)
+                .addSchema(schema)
+                .storeProperties(storeProperties)
+                .build();
+    }
+
+    private Schema resolveSchema(final User user, final GraphLibrary graphLibrary, final String parentSchemaId) {
+        if (isAuthorised(user, idAuths.get(parentSchemaId))) {
+            Schema schema = null;
+            if (parentSchemaId != null) {
+                schema = graphLibrary.getSchema(parentSchemaId);
+            }
+            return schema;
+        } else {
+            throw new IllegalArgumentException("User is not authorised to export using parentSchemaId: " + parentSchemaId);
+        }
+    }
+
+    private StoreProperties resolveStoreProperties(final User user, final GraphLibrary graphLibrary, final String parentStorePropertiesId) {
+        if (isAuthorised(user, idAuths.get(parentStorePropertiesId))) {
+            StoreProperties storeProperties = null;
+            if (parentStorePropertiesId != null) {
+                storeProperties = graphLibrary.getProperties(parentStorePropertiesId);
+            }
+            return storeProperties;
+        } else {
+            throw new IllegalArgumentException("User is not authorised to export using parentStorePropertiesId: " + parentStorePropertiesId);
+        }
+    }
+
+    private static void validate(final String graphId,
+                                 final String parentSchemaId,
+                                 final String parentStorePropertiesId,
+                                 final GraphLibrary graphLibrary,
+                                 final Store store) {
+
+        final ValidationResult result = new ValidationResult();
+
+        if (store.getGraphId().equals(graphId)) {
+            result.addError("Cannot export to the same Graph: " + graphId);
+        }
+        if (null == graphLibrary) {
+            // GraphLibrary is required as only a graphId, a parentStorePropertiesId or a parentSchemaId can be given
+            result.addError("Store GraphLibrary is null");
+        }
+        if (graphLibrary.exists(graphId)) {
```

Comment:
This should be "else if" as if graphLibrary is null then currently you would get a null pointer exception here.